aiida/cmdline/commands/cmd_verdi.py

```diff
@@ -38,22 +38,36 @@
 )
 
 
-class MostSimilarCommandGroup(click.Group):
-    """
-    Overloads the get_command to display a list of possible command
-    candidates if the command could not be found with an exact match.
+class VerdiCommandGroup(click.Group):
+    """Command group for verdi commands.
+
+    Overloads the get_command to
+
+      * display a list of possible command candidates if the command is not an exact match
+      * adds an option to control verbosity
+
+    Note: click has no built-in support for adding common options to subcommands, that's why we hack it in this way.
+    See https://github.com/pallets/click/issues/108 for discussions on this topic.
     """
 
     def get_command(self, ctx, cmd_name):
         """
         Override the default click.Group get_command with one giving the user
         a selection of possible commands if the exact command name could not be found.
         """
-        cmd = click.Group.get_command(self, ctx, cmd_name)
+        cmd = super().get_command(ctx, cmd_name)
 
         # return the exact match
+
         if cmd is not None:
-            return cmd
+
+            # add verbosity option to all commands (but don't add it twice)
+            # Note: this is needed if `get_command` is called more than once on the same command
+            # within the same cli execution (e.g. `verdi help` does this).
+            if '_has_verbosity' in cmd.__dict__:
+                return cmd
+            cmd._has_verbosity = True  # pylint: disable=protected-access
+            return options.VERBOSITY()(cmd)
```

Comment:
using `in cmd.__dict__` is not really pythonic I think. Maybe the following would be better:
```
if hasattr(cmd, '_verbosity_option'):
    return cmd
setattr(cmd, '_verbosity_option', True)
```
but still, maybe I would prefer using the actual `click` interface:
```
if 'verbosity' in [param.name for param in cmd.params]:
    return cmd
return options.VERBOSITY()(cmd)
```

downside is that it is a bit fragile with respect to name changes of the `--verbosity` flag, but since that is unlikely to happen, I think it is acceptable.

Reply:
added in 08ab9da8a017a03372f54d4e8b642a48941e5bad